### PR TITLE
Add view command and NRIC field

### LINKED ISSUE #48: Add View Command
Add view command that allows users to view all in-depth contact and medical information of the patient using the patient's unique NRIC.

### PR DESCRIPTION
**Pull request by Dominic-Khoo**
Add view command that allows user to view all in-depth contact and medical information of the patient using patient's NRIC.

CHANGELOG: 

> Add NRIC field to Person model (together with NricMatchesPredicate class)
> Change add and edit command to support addition of NRIC field
> Add ViewCommandParser to ensure NRIC is entered in appropriate format. NRIC entered also case insensitive.
> Currently, if user is found, the code right now reuses code similar to FindCommand (it will just display the user details in list form)
> Exception thrown if no user found and/or invalid command format is input by the user
> JUnit test cases edited to include testing for ViewCommand and addition of NRIC field

closes #48

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/ViewCommand.java
Comment (itsnotsherm) [inline src/main/java/seedu/address/logic/commands/ViewCommand.java:1]: neat implementation

File: src/main/java/seedu/address/logic/parser/ViewCommandParser.java
Comment (itsnotsherm) [inline src/main/java/seedu/address/logic/parser/ViewCommandParser.java:1]: nice, javadocs can add params and return value so that we know what each method accepts and returns

File: src/main/java/seedu/address/model/person/Nric.java
Comment (itsnotsherm) [inline src/main/java/seedu/address/model/person/Nric.java:23]: does this need to be public

File: src/main/java/seedu/address/model/person/Nric.java
Comment (itsnotsherm) [inline src/main/java/seedu/address/model/person/Nric.java:21]: nice use of regex to verify nric

File: src/test/java/seedu/address/logic/commands/ViewCommandTest.java
Comment (itsnotsherm) [inline src/test/java/seedu/address/logic/commands/ViewCommandTest.java:1]: thanks for adding tests for the new command

File: src/test/java/seedu/address/logic/parser/EditCommandParserTest.java
Comment (itsnotsherm) [inline src/test/java/seedu/address/logic/parser/EditCommandParserTest.java:81]: good catch

**Review by itsnotsherm — APPROVED** [review]
LGTM, check through if you want to edit before merging

**Review by reidenong — APPROVED** [review]
Looks good and works on my machine, unit tests are pretty comprehensive as well. nice one

**Pull request merged**
